### PR TITLE
fix base_dmg and def.dmg_mult being nil

### PR DESCRIPTION
**Pull request by Elkien3**

### CODE REVIEW

File: api.lua [hunk 1]
```diff
@@ -115,7 +115,7 @@ local function fire(stack, player)
 	if pointed and pointed.type == "object" then
 		local target = pointed.ref
 		local point = pointed.intersection_point
-		local dmg = base_dmg * def.dmg_mult
+		local dmg = def.base_dmg * def.dmg_mult
```
Comment (ClobberXD) [inline api.lua]: `base_dmg` is a constant, with which `def.dmg_mult`(which is a multiplier) is multiplied to get the total damage before calculating headshots and the like. So `base_dmg` needs to be defined somewhere at the top of the file. Personally, I prefer to assign it a value of 1, meaning that each shot will deal a damage of 1 HP if `def.dmg_mult` isn't defined.

**Comment by Elkien3** [comment]
Ah, thought that was just a typo. Though I think its probably better to have the multiplier be a constant (or a minetest setting) then have the guns have their own base_dmg. Whatever works for you.

**Comment by ClobberXD** [comment]
It makes more logical sense to have the base remain constant and the multiplier be modifiable. :)

**Comment by Elkien3** [comment]
A multiplier is usually an "instrument or device for multiplying or intensifying some effect."
Not typically for bringing the base value up to what it should be, as in the case of a base_dmg of 1 and a def.dmg_multi of 4. I guess mathematically it's the same thing, but it just seems backwards to me, lol.

**Comment by Elkien3** [comment]
A multiplier is usually "an instrument or device for multiplying or intensifying some effect"
Not typically something that brings a value up to what it normally should be, as in the case of a base_dmg of 1 and a def.dmg_multi of 4.
I guess Mathematically it's the same thing, but it just seems backwards to me, lol.

**Comment by ClobberXD** [comment]
Re-opened. Close if you want to :P

Also, you still need to add `base_dmg = 1` at the top. I'll merge once that's done.

**Comment by ClobberXD** [comment]
562be06

**Pull request closed without merging**